parlai/mturk/core/dev/mturk_manager.py

```diff
@@ -1598,22 +1604,37 @@ def get_assignments_for_hit(self, hit_id):
         assignments_info = client.list_assignments_for_hit(HITId=hit_id)
         return assignments_info.get('Assignments', [])
 
+    # FIXME move this into mturk_utils
+    def check_hit_status(self, hit_id):
+        """Checks the status of a HIT, removes it from outstanding if it is
+        no longer acceptable or pending.
+        """
+        hit = self.get_hit(hit_id)
+        hit_data = hit['HIT']
+        if hit_data['HITStatus'] in ['Reviewable', 'Reviewing', 'Disposed']:
+            self.hit_id_list.remove(hit_id)
+
     def expire_all_unassigned_hits(self):
-        """Move through the whole hit_id list and attempt to expire the
+        """Move through the whole unassigned hit list and attempt to expire the
         HITs, though this only immediately expires those that aren't assigned.
         """
-        # TODO note and mark assigned hits as ones to be expired later.
-        # this will improve the shutdown experience
         shared_utils.print_and_log(
             logging.INFO,
             'Expiring all unassigned HITs...',
             should_print=not self.is_test,
         )
-        completed_ids = self.worker_manager.get_complete_hits()
         for hit_id in self.hit_id_list:
-            if hit_id not in completed_ids:
-                # TODO get confirmation that the HIT is acutally expired
-                mturk_utils.expire_hit(self.is_sandbox, hit_id)
+            mturk_utils.expire_hit(self.is_sandbox, hit_id)
+
+    def expire_all_hits(self):
```

Comment:
how does this function differ from the one directly above it? they appear to be removing from the same `self.hit_id_list`

Reply:
Definitely a bug - fixing!